appcelerator/titanium_mobile
# fix(android): can't set versionCode in manifest as of 9.0.0

**JIRA:**
https://jira.appcelerator.org/browse/TIMOB-27705

**Note 1:**
This issue was never in a release version. Was caught before release.
Regression caused by changing build system from "ant" to "gradle".

**Note 2:**
Android gradle build system ignores "versionCode" and "versionName" set in "AndroidManfiest.xml". This is by Google's design. You are expected to set these settings in the "build.gradle" file instead. This PR maintains backward compatibility by reading these 2 version settings in the "tiapp.xml" and "./platform/android/AndroidManifest.xml" and writing them to the generated "build.gradle" file.

**Note 3:**
Titanium app developer can also set "versionCode" and "versionName" via their own "./platform/android/build.gradle" file. This will take priority over the Titanium generated "build.gradle" file.

---
**Test:**
1. Create a "Classic" Titanium app project.
2. Set up the "tiapp.xml" file with the below.
3. Build for Android.
4. Go to folder: `./build/android/app/build/intermediates/merged_manifests/debug`
5. Open the "AndroidManfiest.xml" file.
6. Verify `android:versionCode` is set to `5`.
7. Copy the below "build.gradle" content to project folder `./platform/android`.
8. Build for Android.
9. Go to folder: `./build/android/app/build/intermediates/merged_manifests/debug`
10. Verify "AndroidManifest.xml" file's `android:versionCode` is set to `7`.

tiapp.xml
```xml
<?xml version="1.0" encoding="UTF-8"?>
<ti:app xmlns:ti="http://ti.appcelerator.org">
	<android xmlns:android="http://schemas.android.com/apk/res/android">
		<manifest android:versionCode="5"/>
	</android>
</ti:app>
```

./platform/android/build.gradle
```groovy
android {
	defaultConfig {
		versionCode 7
	}
}
```
